gnucash/gnome/window-reconcile.c

```diff
@@ -274,78 +283,34 @@ recnRecalculateBalance (RecnWindow *recnData)
     GtkAction *action;
 
     account = recn_get_account (recnData);
-    if (!account)
-        return gnc_numeric_zero ();
+    if (!account) return gnc_numeric_zero ();
 
     reverse_balance = gnc_reverse_balance(account);
-
-    /* update the starting balance */
     include_children = xaccAccountGetReconcileChildrenStatus(account);
-    starting = gnc_ui_account_get_reconciled_balance_as_of_date
-        (account, recnData->statement_date, include_children);
     print_info = gnc_account_print_info (account, TRUE);
 
-    /*
-     * Do not reverse the balance here.  It messes up the math in the
-     * reconciliation window.  Also, the balance should show up as a
-     * positive number in the reconciliation window to match the positive
-     * number that shows in the register window.
-     */
-
-    amount = xaccPrintAmount(starting, print_info);
-    gnc_set_label_color(recnData->starting, starting);
-    gtk_label_set_text(GTK_LABEL(recnData->starting), amount);
-    if (reverse_balance)
-        starting = gnc_numeric_neg (starting);
-
-    /* update the statement date */
-    amount = qof_print_date(recnData->statement_date);
-    gtk_label_set_text(GTK_LABEL(recnData->recn_date), amount);
-
-    /* update the ending balance */
-    ending = recnData->new_ending;
-    if (reverse_balance)
-        ending = gnc_numeric_neg (ending);
-    amount = xaccPrintAmount(ending, print_info);
-    gnc_set_label_color(recnData->ending, ending);
-    gtk_label_set_text(GTK_LABEL(recnData->ending), amount);
-    if (reverse_balance)
-        ending = gnc_numeric_neg (ending);
+    starting = gnc_ui_account_get_reconciled_balance_as_of_date
+        (account, recnData->statement_date, include_children);
```

Comment:
I think it would be clearer to call 
```
label_set_amount (recnData->starting, starting, print_info, FALSE);
```
rather than potentially negating it twice.